[base] Removes invisible tabbable elements from `TrapFocus` component (#33543)
mui/material-ui: docs/data/base/components/trap-focus/trap-focus.md

The following demo uses the [`Portal`](/base/react-portal/) component to render a subset of the `TrapFocus` children into a new "subtree" outside of the current DOM hierarchy, so they are no longer part of the focus loop:

{{"demo": "PortalTrapFocus.js"}}

### Using a toggle inside the trap

> The most common use case for the `TrapFocus` component is to maintain focus within a [`Modal`](/base/react-modal/) component that is entirely separate from the element that opens the modal.

Reviewer: This should be either "modal" or `ModalUnstyled`. I would go with the former since it refers to the generic concept of the component, and the user might not necessarily use the MUI Base `ModalUnstyled`.

```suggestion
The most common use case for the `TrapFocus` component is to maintain focus within a [modal](/base/react-modal/) component that is entirely separate from the element that opens the modal.
```

Author: Ah, great point! I was mixing up `@mui/base/UnstyledModal` with `@mui/material/Modal`

I'll go with the recommendation then!

The following demo uses the [`Portal`](/base/react-portal/) component to render a subset of the `TrapFocus` children into a new "subtree" outside of the current DOM hierarchy, so they are no longer part of the focus loop:

{{"demo": "PortalTrapFocus.js"}}

> ## Using a toggle inside the trap

Reviewer: Let's make this an h3 so it's nested under the Customization h2.

```suggestion
### Using a toggle inside the trap
```

Author: Ah, fantastic catch! Looks like that [changed since I opened this PR](https://github.com/mui/material-ui/commit/d207a3b734c0130248179b5d88a5701b123f68c5#diff-d58236350be1411fc5eaaf3d684738c9d7a5e38f95b0b16c58edc97e9fceb95cR63), and I missed that when I merged 😅

> The most common use case for the `TrapFocus` component is to maintain focus within a [`Modal`](/base/react-modal/) component which is entire separete from the element which would open the modal. However, there are use cases where you may want to have a toggle button for the `open` prop of the `TrapFocus` component which is stored inside that component.

Reviewer: ```suggestion
The most common use case for the `TrapFocus` component is to maintain focus within a [modal](/base/react-modal/) component that is entirely separate from the element that opens the modal. 
But you can also create a toggle button for the `open` prop of the `TrapFocus` component that is stored inside of the component itself, as shown in the following demo:
```

Author: Done, except for one thing. I think it should be "[`Modal`](/base/react-modal/)" and not "[modal](/base/react-modal/)" because directly above this section, the [Portal component is referenced](https://github.com/mui/material-ui/blob/master/docs/data/base/components/trap-focus/trap-focus.md?plain=1#L84) in this exact same way. 😄
